projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -189,32 +195,43 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
     this.onChange = undefined;
   }
 
+  // when getDefaultValues is defined, you do not need to specify the default values
+  // in your form (the ones defined within the `getFormControls` method)
+  protected getDefaultValues(): Partial<FormInterface> | undefined {
+    return undefined;
+  }
+
   public writeValue(obj: Required<ControlInterface> | null): void {
     // @hack see where defining this.formGroup to undefined
     if (!this.formGroup) {
       return;
     }
 
+    const defaultValues: Partial<FormInterface> | undefined = this.getDefaultValues();
+
     // should accept falsy values like `false` or empty string
     // if the value is null or undefined it might be because we're
     // switching from one value of a polymorphic type to another
     // for ex and in that case we don't want to go further
     if (isNullOrUndefined(obj)) {
-      return;
-    }
-
-    const transformedValue: FormInterface = this.transformToFormGroup(obj);
+      if (this.onChange) {
+        this.onChange(null);
```

Comment:
is this actually needed now? I feel like the form control will already be set to null?

Reply:
Yes it's still needed, but there's one thing I'm not sure of: Should we have `emitEvent` set to true below :think:

Reply:
Forgot to update but I think you're right and it's not needed :)